foodnet/common/middleware.py

```diff
@@ -0,0 +1,23 @@
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import reverse
+from django.contrib import messages
+
+from foodnet.membership.models import UserProfile
+
+
+class NewUserForceProfileMiddleware(object):
+    def process_request(self, request):
+        allowed_paths = (
+            reverse('profile'),
+            reverse('account_login'),
+            reverse('account_logout'),
+            reverse('new_member_set_password'),
+        )
+        if request.user.is_authenticated() and \
+                not request.user.is_superuser and \
+                request.path not in allowed_paths:
```

Comment:
Another formatting tip:
```
        if (
            request.user.is_authenticated()
            and not request.user.is_superuser
            and request.path not in allowed_paths
        ):
```

PEP8 says that parentheses are to be preferred for formatting long lines (https://www.python.org/dev/peps/pep-0008/#maximum-line-length)